docs/triggers.md

```diff
@@ -51,6 +51,28 @@ spec:
     ref: pipeline-template
 ```
 
+Triggers can reference templates and bindings based on data in the event body, headers, or processed extensions. This allows contextual information to be pulled in from the event to determine which binding or template should be populated. The `ref` field for `bindings` and `templates` can be populated in the same manner as triggertemplate params.
+
+This can be useful when different pipelines require different resources or workspaces to run. `TriggerTemplate` objects by themselves must have a static configuration structurually, even if individual strings are able to be populated.
+
+<!-- FILE: examples/triggers/trigger-with-event-reference.yaml -->
+```YAML
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: trigger
+spec:
+  interceptors:
+    - cel:
+        overlays:
+        - key: extensions.truncated_sha
+          expression: "body.pull_request.head.sha.truncate(7)"
+  bindings:
+  - ref: pipeline-$(body.repository.owner.login)-binding
+  template:
+    ref: pipeline-$(header.X-GitHub-Event)-trigger
```

Comment:
So someone will still have to create a bunch of templates and bindings with valid values of `X-Github-Event` and `repository.owner.login` ?

Assuming we have a solution the reusable interceptor processing problem, one potential alternative could be having all the interceptors in a chain and then each trigger has a simple CEL filter plus a static binding and template value. 

It is a bit more verbose but IMO makes things the trigger config a bit easier to parse. What do you think?

Reply:
I think that this formatting still makes sense when you want flexibility based off extension values. Remember one of the key things we are trying to enable here is workspace configuration that differs based off metadata on the input. I think that static binding and template still have the issue specifically around static triggertemplates.

Bindings have more flexibility for population of a range of values since triggerbindings can have more parameters than the template itself. What I have found using this capability is that I have to worry far less about defaults because I specify exactly the set of extensions I am looking for in a triggerbinding. Now, a goal here is not performance improvement, but rather ease the operation and management of a tekton platform.

The `resolve` operations still operate in the same manner. Triggerbindings and templates resolve within the namespace of the Trigger that invoke them.

I can see a concern around the platform management use case: "What will this trigger do when I invoke it" is an important question. I think improving the story about logging and debugging (as @bigkevmcd has in #954) is the biggest gain here. Personally, I find these configurations still readable, as they parse similarly to TriggerBinding values when I read through them.